lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -381,9 +459,38 @@ internal int NumberOfAppErrors()
         /// Checks if all the tasks are in final states
         /// </summary>
         /// <returns></returns>
-        internal bool AllInFinalState()
+        internal bool AreAllTasksInFinalState()
+        {
+            var notInFinalState = _tasks.Where(t => !t.Value.TaskState.IsFinalState()).Take(5).ToList();
```

Comment:
We count all tasks which are not in final state yet, but display at most 5 to avoid logs with 100s of tasks listed.